compiler/modules/front/codeGen.bal

```diff
@@ -868,6 +868,12 @@ function codeGenAssignStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environ
 }
 
 function codeGenAssignToVar(CodeGenContext cx, bir:BasicBlock startBlock, Environment env, string varName, s:Expr expr, Position pos) returns CodeGenError|StmtEffect {
+    var [unnarrowedReg, assignments] = check assignmentRegister(cx, env, varName, pos);
+    bir:BasicBlock nextBlock = check codeGenAssign(cx, env, startBlock, unnarrowedReg, expr, unnarrowedReg.semType, pos);
+    return { block: nextBlock, assignments };
+}
+
+function assignmentRegister(CodeGenContext cx, Environment env, string varName, Position pos) returns CodeGenError|[bir:Register, int[]] {
```

Comment:
I would call this something like `lookupVarRefForAssign`

Reply:
fixed